### PR TITLE
Install Ocaml toolchain on CircleCI

### PR DESCRIPTION
**Pull request by v-jizhang**
Install Ocaml toolchain on CircleCI
Test should pass:
./bin/buck test //test/com/facebook/buck/features/ocaml/...

### CODE REVIEW

File: test/com/facebook/buck/features/ocaml/OCamlIntegrationTest.java [hunk 1]
```diff
@@ -229,7 +229,7 @@ public void testNativePlugin() throws Exception {
     workspace.runBuckCommand("build", binTarget.toString()).assertSuccess();
 
     Path ocamlNativePluginDir =
-        workspace.getDestPath().resolve("buck-out").resolve("gen").resolve("ocaml_native_plugin");
+        workspace.getDestPath().resolve("buck-out").resolve("gen").resolve("ce9b6f2e").resolve("ocaml_native_plugin");
```
Comment (zpao) [inline test/com/facebook/buck/features/ocaml/OCamlIntegrationTest.java]: Let's see if we can not hard code this

File: test/com/facebook/buck/features/ocaml/OCamlIntegrationTest.java [hunk 2]
```diff
@@ -517,8 +517,8 @@ public void testConfigWarningsFlags() throws IOException {
     workspace.runBuckCommand("build", target.toString()).assertFailure();
     BuckBuildLog buildLog = workspace.getBuildLog();
     assertTrue(buildLog.getAllTargets().containsAll(targets));
-    buildLog.assertTargetCanceled(target);
-    buildLog.assertTargetCanceled(binary);
+    buildLog.assertTargetFailed(target.toString());
```
Comment (zpao) [inline test/com/facebook/buck/features/ocaml/OCamlIntegrationTest.java]: What's the difference between canceled and failed?

Comment (v-jizhang) [reply test/com/facebook/buck/features/ocaml/OCamlIntegrationTest.java:520]: There is no doc defines them and assertTargetCanceled() is not used anywhere in buck code. My understanding is CANCELED is used internally and the final result in buildLog is always SUCCESS or FAILED.

Comment (v-jizhang) [reply test/com/facebook/buck/features/ocaml/OCamlIntegrationTest.java:520]: All ocaml tests are skipped in Sandcastle.

**Pull request merged**